client/matching/client.go

```diff
@@ -192,18 +192,30 @@ func (c *clientImpl) QueryWorkflow(
 	ctx context.Context,
 	request *types.MatchingQueryWorkflowRequest,
 	opts ...yarpc.CallOption,
-) (*types.QueryWorkflowResponse, error) {
+) (*types.MatchingQueryWorkflowResponse, error) {
 	partition := c.loadBalancer.PickReadPartition(
 		persistence.TaskListTypeDecision,
 		request,
 		"",
 	)
+	originalTaskListName := request.TaskList.GetName()
```

Comment:
Nit: you can copy the whole task list:
```suggestion
	originalTaskList := *request.TaskList
```
Then pass it, avoiding restoring the name.
```
c.provider.UpdatePartitionConfig(
		request.GetDomainUUID(),
		originalTaskList,
		persistence.TaskListTypeDecision,
		resp.PartitionConfig,
```